tests/test_domain.py

```diff
@@ -85,6 +96,113 @@ def test_build_simple_domain():
     assert domain
 
 
+def test_build_simple_domain_with_derived_predicates():
+    """Test a simple PDDL domain with derived predicates."""
+    x, y, z = variables("x y z")
+    p = Predicate("p", x, y, z)
+    q = Predicate("q")
+    r = Predicate("r")
+    dp = DerivedPredicate(p, And(q, r))
+    action_1 = Action("action_1", [x, y, z], precondition=p, effect=Not(p))
+    domain = Domain(
+        "simple_domain",
+        predicates={p},
+        derived_predicates={dp},
+        actions={action_1},
+    )
+    assert domain
+
+
+def test_build_domain_with_numeric_fluents():
+    """Test a PDDL domain with simple numeric fluents."""
+    x, y, z = variables("x y z")
+    p = Predicate("p", x, y, z)
+    q = Predicate("q")
+    r = Predicate("r")
+    func1 = NumericFunction("f1", x, y)
+    func2 = NumericFunction("f2")
+    func3 = NumericFunction("f3")
+    action_1 = Action(
+        "action_1",
+        [x, y, z],
+        precondition=p
+        & FunctionEqualTo(func1, NumericValue(0))
```

Comment:
We should probably overload the expected operators to make the programmatic creation more succinct. Follow-up issue/PR?

Reply:
Definitely! Given the amount of change here I guess it's better to do that in a follow-up PR